src/main/java/seedu/address/logic/commands/RequiredCommand.java

```diff
@@ -0,0 +1,319 @@
+    public static final String MESSAGE_SUCCESS_MATHANDSCI = "You have completed your "
+            + "Math and Science Module Requirement!";
+    public static final String MESSAGE_FAILURE_ITPROF = "There was an error loading the IT Professionalism Modules :(";
+    public static final String MESSAGE_FAILURE_MATHANDSCI = "There was an error loading"
+            + " the Math and Science Modules :(";
+    public static final Path FOUNDATION_PATH = Paths.get("src/main/data/foundationmodules.json");
+    public static final Path INTERNSHIP_PATH = Paths.get("src/main/data/industrialexperience.json");
+    public static final Path ITPROF_PATH = Paths.get("src/main/data/ITProfessionalism.json");
+    public static final Path MATHANDSCI_PATH = Paths.get("src/main/data/mathandsciencemodules.json");
+    public static final Path SCIENCE_PATH = Paths.get("src/main/data/sciencemodules.json");
+
+    private ObservableList<Module> modules;
+    private String leftOverModules = "";
+    private Optional<ReadOnlyGradPad> foundationStorage;
+    private ObservableList<Module> requiredFoundation;
+    private Optional<ReadOnlyGradPad> scienceStorage;
+    private ObservableList<Module> requiredScience;
+    private Optional<ReadOnlyGradPad> internshipStorage;
+    private ObservableList<Module> requiredInternship;
+    private Optional<ReadOnlyGradPad> itProfStorage;
+    private ObservableList<Module> requiredITprof;
+    private Optional<ReadOnlyGradPad> mathAndScienceStorage;
+    private ObservableList<Module> requiredMathAndScience;
+
+    /**
+     * Returns foundationStorage attribute of RequiredCommand object.
+     * @return foundationStorage attribute of type Optional<ReadOnlyGradPad/>.
+     */
+    public Optional<ReadOnlyGradPad> getFoundationStorage() {
+        return foundationStorage;
+    }
+    /**
+     * Loads the foundationStorage attribute with Foundation Modules.
+     * @throws IOException
+     * @throws DataConversionException
+     */
+    public void setFoundationStorage(Path path) throws IOException, DataConversionException {
+        JsonGradPadStorage storage = new JsonGradPadStorage(path);
+        this.foundationStorage = storage.readGradPad();
+    }
+
+    /**
+     * Cross references the user's current list of Modules and marks out
+     * any undone Foundation Modules.
+     */
+    public void compareFoundation() {
+        boolean foundationClear = true;
+        String modulesToAdd = "";
+        for (Module module : requiredFoundation) {
+            boolean add = true;
+            for (Module mod : modules) {
+                if (module.isSameModule(mod)) {
+                    add = false;
+                    break;
+                }
+            } if (add) {
+                String moduleToAdd = module.getModuleCode() + " (" + module.getModularCredits() + " MCs)";
+                modulesToAdd += "\n" + moduleToAdd;
+                foundationClear = false;
+            }
```

Comment:
True! I combined the 3 methods into one (those that compare foundation, ITprof and mathandsci mods) into one method. I decided to keep the compareInternship and compareScience methods as they have slightly different implementations than the other 3 module categories

Reply:
I think Sam meant that the logic behind the implementation can be abstracted out so there's not as much repeated code because now there are still repeated code in the 3 methods which is the double for loops.
I'm guessing maybe the addition of the helper function would be good.

Reply:
oh right HAHAH okay abstracted further